### DISCUSSION TITLE
allow ref var with initial value

### DISCUSSION THREAD

**Discussion by MkazemAkhgary** [dotnet/csharplang#1077] [General]
allow to use ref var with initial or default value.

```
if (!map.AddOrFind(ref var kvp = (item, slot).AsPair(), false);
{
    // ...
}
```

equivalent to

```
var kvp = (item, slot).AsPair();
if (!map.AddOrFind(ref kvp, false))
{
   // ...
}
```

AsPair is extension method to convert tuple to key value pair.

```
public static KeyValuePair<TKey, TValue> AsPair<TKey, TValue>(this (TKey key, TValue value) tuple)
{
    return new KeyValuePair<TKey, TValue>(tuple.key, tuple.value);
}
```

---

edited with real example of what I had, AddOrFind method exist [here](https://github.com/qwertie/ecsharp/blob/master/Core/Loyc.Collections/Sets/MMap.cs)

--- 

**some considerations:** 

 - should this syntax be allowed too?

```
Call(ref var x, y); // x assigns to default(T), var is inferred from method signature if there are no ambiguous overloads.
```

**Comment by jnm2** [comment]
Duplicate of https://github.com/dotnet/csharplang/issues/854
